dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/SqlViewController.java

```diff
@@ -71,9 +81,15 @@
     // -------------------------------------------------------------------------
 
     @RequestMapping( value = "/{uid}/data", method = RequestMethod.GET, produces = ContextUtils.CONTENT_TYPE_JSON )
-    public @ResponseBody Grid getViewJson( @PathVariable( "uid" ) String uid,
-        @RequestParam( required = false ) Set<String> criteria, @RequestParam( required = false ) Set<String> var,
-        HttpServletResponse response ) throws WebMessageException
+    public @ResponseBody RootNode getViewJson(
```

Comment:
@lkwaerst take a look at the ValidationResultController / ValidationResultQuery, could you try to replicate the same way of doing paging as there? Basically it means putting the params into another class, which will keep the controller a little cleaner :)